src/main/java/com/thealgorithms/greedyalgorithms/Huffman.java

```diff
@@ -0,0 +1,77 @@
+// Huffman Coding in Java
+
+import java.util.PriorityQueue;
+import java.util.Comparator;
+
+class HuffmanNode {
+    int item;
+    char c;
+    HuffmanNode left;
+    HuffmanNode right;
+}
+
+// For comparing the nodes
+class ImplementComparator implements Comparator<HuffmanNode> {
+    public int compare(HuffmanNode x, HuffmanNode y) {
+        return x.item - y.item;
+    }
+}
+
+// IMplementing the huffman algorithm
+public class Huffman {

```

Comment:
Looks like a similar class already exists. What's the difference between this?

https://github.com/TheAlgorithms/Java/blob/master/src/main/java/com/thealgorithms/others/Huffman.java

You can refactor the existing one and add tests there. Also it would prefer if tests are parameterized, there are a lot of such examples in project.

Reply:
Ok will add the test cases to the existing one 👍🏻